src/main/java/com/google/devtools/build/lib/rules/cpp/CppFileTypes.java

```diff
@@ -176,19 +176,21 @@ public boolean apply(String path) {
 
   // Matches shared libraries with version names in the extension, i.e.
   // libmylib.so.2 or libmylib.so.2.10 or libmylib.so.1a_b35.
-  private static final Pattern VERSIONED_SHARED_LIBRARY_PATTERN =
-      Pattern.compile("^.+\\.so(\\.\\d\\w*)+$");
+  private static final Pattern VERSIONED_SO_PATTERN = Pattern.compile("^.+\\.so(\\.\\d\\w*)+$");
+  private static final Pattern VERSIONED_DYLIB_PATTERN =
```

Comment:
Not that I can review this, but it looks like these two could be folded into the single
("^.+\\.((so)|(dylib))(\\.\\d\\w*)+$");